dolphin/processor/config.py

```diff
@@ -508,17 +509,25 @@ def get_index_source_light_model_list(self):
          (for multiple filters)
 
         """
-        if 'lens_light' in self.settings['model']:
-            index_source_light_model_list = []
-            index_num = 0
-            for i in range(self.band_number):
-                single_index_list = []
-                for j in range(len(self.settings['model']['source_light'])):
-                    single_index_list.append(index_num)
-                    index_num += 1
-                index_source_light_model_list.append(single_index_list)
-            return index_source_light_model_list
-
+        if 'source_light' in self.settings['model']:
+            if self.band_number == 1:
+                index_list = [[]]
+                for k, model in enumerate(
+                        self.settings['model']['source_light']):
+                    index_list[0].append(k)
+                return index_list
+            else:
+                if 'source_light_band_index' in self.settings['model']:
+                    index_list_settings =\
+                        self.settings['model']['source_light_band_index']
+                    index_list = [[] for _ in range(self.band_number)]
```

Comment:
You can simplify this line as:

```
index_list = [[]] * self.band_number
```

Reply:
That was what I first tried too, but I think that will result in the list containing the same list object N times, i.e. if we add an index in the first list, the second list will also have the same index.

Reply:
O I see, didn't realize that would happen, then the current way is the right way!